vernac/comProgramFixpoint.ml

```diff
@@ -217,6 +223,7 @@ let build_wellfounded pm (recname,pl,bl,arityc,body) poly ?typing_flags ?using r
     sigma, mkApp (h_a_term, [| argtyp ; wf_rel ; h_e_term; prop |])
   in
   let sigma, def = Typing.solve_evars env sigma def in
+  let sigma = Evd.collapse_sort_variables sigma in
```

Comment:
Is this changed linked to a bug?

Reply:
We shouldn't send qualities to the kernel, so anything that goes through Declare should collapse the corresponding evarmap. If you're able to give me a spec of the Program code we may or may not decide that this is the right place to do it.

Reply:
If it fixes a bug then we should add it to the test suite
If  it doesn't then what is it for?

Reply:
Ah, so yes, it does fix a bug, but the example is not reproducible on master. You need the full version of #17125 to trigger it.

Reply:
#18843 may fix the mentioned "bug".